db/memtable_list.cc

```diff
@@ -115,22 +115,26 @@ bool MemTableListVersion::GetFromHistory(const LookupKey& key,
                                          RangeDelAggregator* range_del_agg,
                                          SequenceNumber* seq,
                                          const ReadOptions& read_opts) {
-  return GetFromList(&memlist_history_, key, value, s, merge_context,
-                     range_del_agg, seq, read_opts);
+  PinnableSlice pSlice;
+  PinnableSlice* pSlicePtr = value != nullptr ? &pSlice : nullptr;
+  auto res = GetFromList(&memlist_history_, key, pSlicePtr, s, merge_context,
+                         range_del_agg, seq, read_opts);
+  if (value != nullptr) {
```

Comment:
if we are doing this check anyway, can we rewrite the code to be something like this
```
if (LIKELY(value != nullptr)) {
  PinnableSlice pinnable_val;
  res = GetFromList( .... );
  value->assign(pinnable_val.data(), pinnable_val.size());
} else {
  res = GetFromList( .... );
}
```

What do you think ?